### ISSUE TITLE
Search failed error with version 0.1.5

### ISSUE THREAD

**Issue by fmcpherson** [fedwiki/wiki-plugin-search#15]
I've updated my install of wiki and now running the latest. I test Federation Search via the new hamburger menu and when I do I immediately see search failed: error. 

I observe that the package.json in node_modules/wiki-plugin-search says this is at version 0.1.5 

I have another instance of wiki installed on my local computer that I have not yet updated and search is working as expected on it. When I look at package.json in this computer is see the version is at 0.1.4. This suggests something in the new version of this plugin breaks it.

**Comment by WardCunningham** [comment]
I'm not able to duplicate this error. I'm using the current install which includes the version of search you mentioned. The error message comes from here:
https://github.com/fedwiki/wiki-plugin-search/blob/b3778326b4fabaa9d9f23c9b3d497dacfb9b7f77/client/search.coffee#L55
If this happens again you might look at the javascript console or network tabs for additional information.

**Comment by paul90** [comment]
Adding some details from https://github.com/fedwiki/wiki/issues/123

The search request is getting blocked as mixed content by the browser. 
![Screenshot 2019-04-19 at 07 16 28](https://user-images.githubusercontent.com/1552489/56410466-1be1f680-6275-11e9-8b5d-3223c7f2ae91.png)
The only way of resolving this issue for those running wiki using TLS would be to have search using a `https://` url.

**Comment by WardCunningham** [comment]
Of course. The browser disallows this and won't provide an explanation to javascript other than "error". Why? I can only assume there is some imagined threat that this avoids. Our democratic way of life is thus protected. We have a partial solution to this that the search plugin should employ. I'll work on a pull request.

**Comment by kevinrutherford** [comment]
I'm seeing this too, since I added TLS to my site. Happy to help with that pull request if I can...

**Comment by WardCunningham** [comment]
I'm not sure if there are multiple issues here. I understand that one complication is that the search service I run is accessed via http and that this is disallowed in some browser circumstances.

A common workaround is to proxy http requests for tls sites. We offer a closed proxy to logged in user but not one that supports the search protocol or isn't properly engaged by the search plugin. We could work to resolve this or we could add a server-side component to the search plugin that would proxy these specific requests.

One might suggest that I just serve search requests via https. I resist this on both practical and philosophical grounds. Practically search performance we now have depends on solid-state disk and ample quiescent memory for kernel buffers. Any further administrative tasks here have a low priority. My philosophical objection has many aspects but one is that I don't think I should have to seek credentials to from any additional bodies, especially ones of questionable character, in order to serve the internet.
